src/main/java/net/dv8tion/jda/api/interactions/components/buttons/Button.java

```diff
@@ -158,7 +167,8 @@ default Button withDisabled(boolean disabled)
     @CheckReturnValue
     default Button withEmoji(@Nullable Emoji emoji)
     {
-        return new ButtonImpl(getId(), getLabel(), getStyle(), getUrl(), isDisabled(), emoji);
+        Checks.check(getStyle() != ButtonStyle.PREMIUM, "Premium components cannot have emojis");
```

Comment:
This is undocumented, also need the same for `label`, `url`, and `id`. Also, this check only matters if `emoji` is not null.

Maybe just add a `checkValid()` in `ButtonImpl` to do this? Then we can also test that check method separately from these.